app/cdap/components/NamespaceAdmin/SourceControlManagement/SourceControlManagementModal.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+import { ISourceControlManagementConfig } from './types';
+import { SourceControlManagementForm } from './SourceControlManagementForm';
+
+const ModalRoot = styled.div`
+  position: absolute;
+  top: 50px;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  z-index: 1061;
+  text-align: left;
+  background-color: white;
+  overflow: auto;
+`;
+
+interface ISourceControlManagementModalProps {
+  isOpen: boolean;
+  toggleModal: () => void;
+  onSubmit: (formState: ISourceControlManagementConfig) => void;
+  sourceControlManagementConfig?: ISourceControlManagementConfig;
+}
+
+const SourceControlManagementModal = ({
+  isOpen,
+  toggleModal,
+  onSubmit,
+  sourceControlManagementConfig,
+}: ISourceControlManagementModalProps) => {
+  return (
+    <>
+      {isOpen && (
+        <ModalRoot>
```

Comment:
Why not use one of the existing modals/dialogs, like Mui `Dialog`?

Reply:
The form will be fullscreen, not exactly a dialog. I didn't find any existing components to use